content/develop/start/hardhat.md

```diff
@@ -3,3 +3,241 @@ title: "Aurora: Getting Started with Hardhat"
+echo "AURORA_PRIVATE_KEY=YOUR_AURORA_PRIVATE_KEY_HERE" >> .env
+yarn install
+```
+
+## Deploy ERC20
+
+The ERC20 example is about a naive Watermelon token 🍉. You can exchange
+them into actual Watermelons 🍉🍉🍉. The total supply is `1000000`, the
+minter is the contract deployer address, and the decimals are `0`
+(One token --> One watermelon).
+
+To deploy the `ERC20` token contract, use the following command:
+
+```bash
+$ make deploy NETWORK=testnet_aurora
+yarn hardhat run scripts/deploy.js --network testnet_aurora
+yarn run v1.22.10
+Deploying contracts with the account: 0x6A33382de9f73B846878a57500d055B981229ac4
+Account balance: 2210010200000000000
+WatermelonToken deployed to: 0xD7f2A76F5DA173043E6c61a0A18D835809A07766
+✨  Done in 14.96s.
+```
+
+## Hardhat Tasks
+
+Hardhat tasks take care of parsing the values provided for each parameter.
+It gets the values, performs the type validation and converts them into your desired type.
+
+In this example, we will go through a set of pre-defined hardhat tasks
+that uses the Hardhat Runtime Environment ([HRE](https://hardhat.org/advanced/hardhat-runtime-environment.html)). In order to complete the tutorial,
+you should use them in the same order:
+
+### ETH Balance
+
+The following Hardhat task uses the `Web3` plugin to get the account’s balance:
+
+```javascript
+task("balance", "Prints an account's balance")
+  .addParam("account", "The account's address")
+  .setAction(async taskArgs => {
+    const account = web3.utils.toChecksumAddress(taskArgs.account);
+    const balance = await web3.eth.getBalance(account);
+
+    console.log(web3.utils.fromWei(balance, "ether"), "ETH");
+  });
+```
+
+To get the `ETH` balance, use the following command:
+
+```bash
+npx hardhat balance --network testnet_aurora --account 0x6A33382de9f73B846878a57500d055B981229ac4
+2.2100102 ETH
+```
+
+You should notice that `--network` is a global built-in option (parameter)
+in Hardhat. We will use it for the following commands as well.
+
+### Total Supply
+
+The following task script gets the total supply of the Watermelon ERC20 token.
+First it attachs the
+token contract, gets the sender address and finaly retrieves the total supply
+by calling `totalSupply()` method in our ERC20 contract. The `--token`
+address is the ERC20 contract address.
+
+```javascript
+task("totalSupply", "Total supply of ERC20 token")
+.addParam("token", "Token address")
+.setAction(async function ({ token }, { ethers: { getSigners } }, runSuper) {
+  const watermelonToken = await ethers.getContractFactory("WatermelonToken")
+  const watermelon = watermelonToken.attach(token)
+  const [minter] = await ethers.getSigners();
+  const totalSupply = (await (await watermelon.connect(minter)).totalSupply()).toNumber()
+  console.log(`Total Supply is ${totalSupply}`);
+});
+```
+
+To get the `totalSupply`, use the following command:
+
+```bash
+$ npx hardhat totalSupply --token 0xD7f2A76F5DA173043E6c61a0A18D835809A07766 --network testnet_aurora
```

Comment:
I think it would be nice to add a step where we define a `TOKEN_ADDRESS` environment variable from the `deploy` output and use that in all these commands. It's a little annoying to change your token's address to my token's address in each of these commands.